msm/src/serialization/witness.rs

```diff
@@ -17,9 +20,26 @@ pub struct Env<Fp> {
     /// field Kimchi gate
     pub intermediate_limbs: [Fp; N_INTERMEDIATE_LIMBS],
 
+    /// Keep track of the RangeCheck4 lookup multiplicities
     // Boxing to avoid stack overflow
     pub lookup_multiplicities_rangecheck4: Box<[Fp; 1 << 4]>,
+
+    /// Keep track of the RangeCheck4 table multiplicities.
+    /// The value `0` is used as a (repeated) dummy value.
+    // Boxing to avoid stack overflow
+    pub lookup_t_multiplicities_rangecheck4: Box<[Fp; 1 << 4]>,
+
+    /// Keep track of the RangeCheck15 lookup multiplicities
+    /// No t multiplicities as we do suppose we have a domain of
```

Comment:
How does this work? You still need to construct the multiplicities for `t`, I guess you wanted to say that it's easier in the case of 2^15 domain size, but why? :thinking:

Reply:
See equation (14): https://eprint.iacr.org/2022/1530.pdf

Reply:
I thought (14) does not apply to our case because our `t(x)` is fully injective, since every element in the range table is unique?